packages/malloy/src/malloy.ts

```diff
@@ -908,11 +901,15 @@ export class PreparedQuery implements Taggable {
    * @return A fully-prepared query (which contains the generated SQL).
    */
   public get preparedResult(): PreparedResult {
+    return this.getPreparedResult();
+  }
+
+  public getPreparedResult(options?: PrepareResultOptions): PreparedResult {
     const queryModel = new QueryModel(this._modelDef);
     const translatedQuery = queryModel.compileQuery(
       this._query,
       false,
-      this.replaceMaterializedReferences
+      options?.replaceMaterializedReferences ?? false
```

Comment:
Getting a `PreparedResult` from a `PreparedQuery` involves running the compiler, so you may want to pass in `replaceMaterializedReferences`.